### ISSUE TITLE
Ninjia builder no longer works on Windows after MSYS2 update

### ISSUE THREAD

**Issue by Mark-64** [darktable-org/darktable#9745]
Trying to build DT git master on Windows.
After updating MSYS2, Ninja builder no longer works. It worked fine before MSYS2 update.
Build fails with this error
```
[59/733] Linking C static library bin\external\libxcf\libxcf.a
FAILED: bin/external/libxcf/libxcf.a 
cmd.exe /C "cd . && C:\msys64\mingw64\bin\cmake.exe -E rm -f bin\external\libxcf\libxcf.a && C:\msys64\mingw64\bin\ar.exe qc bin\external\libxcf\libxcf.a -Wl,--no-as-needed bin/external/libxcf/CMakeFiles/xcf.dir/xcf.c.obj bin/external/libxcf/CMakeFiles/xcf.dir/xcf_names.c.obj && C:\msys64\mingw64\bin\ranlib.exe bin\external\libxcf\libxcf.a && cd ."
C:\msys64\mingw64\bin\ar.exe: invalid option -- W
```
Also, `CMakeError.log` has weird errors
[CMakeError.log](https://github.com/darktable-org/darktable/files/6946728/CMakeError.log)

`-G "MSYS Makefiles"` generator works fine

**Comment by johnny-bit** [comment]
Noticed that earlier with CI. it also fails on mac. For some reason ninja wants to pass linker args to ar. There's nothing in libxcf cmakelist.txt that would point to such problems as it's clear-cut: https://github.com/houz/libxcf/blob/master/CMakeLists.txt

The error must be imo either with CMake or Ninja or both.

since all i had was CI logs I couldn't really fill any bug reports to neither cmake nor ninja... Since you seem to have everything you need - can you please fill the bug report with cmake and/or ninja and link here please?

**Comment by Mark-64** [comment]
@johnny-bit It is clearly a cmake issue.
From ` /var/log/pacman.log`, I see that ninja has not been updated recently, while
`[2021-08-06T17:22:51+0200] [ALPM] upgraded mingw-w64-x86_64-cmake (3.20.5-2 -> 3.21.1-1)`
Downgrading to the previous version, it builds fine.
So I will file a bug with cmake only.

**Comment by Mark-64** [comment]
[Here](https://gitlab.kitware.com/cmake/cmake/-/issues/22520) is the issue. I noticed there is [another regression](https://gitlab.kitware.com/cmake/cmake/-/issues/22512), maybe related.

**Comment by Mark-64** [comment]
It seems a known bug, already closed in the upcoming CMake 3.21.2

**Comment by johnny-bit** [comment]
Great! Thank you! seems like we can close this then and soon after 3.21.2 release add ninja back to CI runs :)
